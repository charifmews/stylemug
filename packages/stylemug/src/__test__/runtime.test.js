import runtime from '../index';

describe('runtime', () => {
  let styles;

  beforeEach(() => {
    styles = runtime.create({
      foo: {
        key1: 'className1',
        key2: 'className2',
      },
      bar: {
        key1: 'className3',
      },
    });
  });

  it('should resolve multiple class names', () => {
    const classNames = styles('foo', 'bar');
    expect(classNames).toBe('className3 className2');
  });

  it('should resolve falsy lookups (first arg)', () => {
    const classNames = styles(false && 'foo', 'bar');
    expect(classNames).toBe('className3');
  });

  it('should resolve falsy lookups (second arg)', () => {
    const classNames = styles('foo', false && 'bar');
    expect(classNames).toBe('className1 className2');
  });

  describe('warnings', () => {
    beforeEach(() => {
      global.console.warn = jest.fn();
    });

    it('should warn when a lookup failed', () => {
      const spy = jest.spyOn(console, 'warn');
      const classNames = styles('unknown');

      expect(spy).toHaveBeenCalledWith(
        '[stylemug] The class name "unknown" does not exist in your stylesheet. Check your stylemug.create({}) definition.'
      );

      spy.mockRestore();
    });
  });
});
