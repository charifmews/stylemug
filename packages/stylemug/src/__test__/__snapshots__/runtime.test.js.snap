// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtime resolver should expose classnames as properties on resolver 1`] = `
Array [
  Object {
    "key1": "className1",
    "key2": "className2",
  },
  Object {
    "key1": "className3",
  },
]
`;
